deployer/deployer-services/src/main/java/io/hyscale/deployer/services/util/K8sPodUtil.java

```diff
@@ -189,6 +189,21 @@ public static boolean checkForPodCreation(V1Pod v1Pod) {
         }
         return false;
     }
+    
+    
+	public static boolean checkForPodFalied(V1Pod v1Pod) {
+		for (V1ContainerStatus v1conContainerStatus : v1Pod.getStatus().getContainerStatuses()) {
+			String reason = null;
+			if (v1conContainerStatus.getState().getWaiting() != null) {
+				reason = v1conContainerStatus.getState().getWaiting().getReason();
+			}
+			if (K8SRuntimeConstants.POD_ERROR_SATES.contains(reason)) {
+				return true;
+			}
+		}
+		return false;
```

Comment:
1.GetAggregatedStatus of each Pod  
2.GetPodStatus -> PodStatus
3.see if it failed is true

This makes sure that we are handling all the error states and it will be reliable